packages/lint-a11y/src/index.ts

```diff
@@ -0,0 +1,237 @@
+import { type Plugin, type ParsedToken, type LintNotice, type ParsedColorToken, type ParsedTypographyToken } from '@cobalt-ui/core';
+import { blend, modeRgb, modeHsl, modeHsv, modeP3, modeOkhsl, modeOklch, modeOklab, modeXyz50, modeXyz65, modeLrgb, useMode, wcagContrast } from 'culori/fn';
+import { APCAcontrast, sRGBtoY } from 'apca-w3';
+
+// register colorspaces for Culori to parse (these are side-effect-y)
+useMode(modeHsl);
+useMode(modeHsv);
+useMode(modeLrgb);
+useMode(modeOkhsl);
+useMode(modeOklab);
+useMode(modeOklch);
+useMode(modeP3);
+const toRgb = useMode(modeRgb);
+useMode(modeXyz50);
+useMode(modeXyz65);
+
+export const RULES = {
+  contrast: 'a11y/contrast',
+};
+
+export const WCAG2_MIN_CONTRAST = {
+  AA: 4.5,
+  AAA: 7,
+};
+
+/**
+ * Table indexed by font size, then by weight ([100,200,300,400,500,600,700,800,900]).
+ * - First, start by finding the row where the the index is smaller than the given font.
+ * - Next, start with the font width, rounded DOWN to the nearest 100.
+ * A value of `null` means it’s innaccessible (out of readable range)
+ * @see https://www.myndex.com/APCA/
+ */
+export const APCA_FONT_LOOKUP_TABLE = [
```

Comment:
TODO: this still seems like a very “loose” guideline by APCA. There’s mention of “bronze,” “silver,” and “gold” compliance, but the actual guidelines based on font size + weight are pretty loose.

I was planning on the following workflow:

1. You (the user) have a color + typography set you want to test in APCA. You then specify `"bronze"`/`"silver"`/`"gold"` based on what you want.
2. If you pass it, great! Good job.
3. If you failed, the error message would say _by how much_ (e.g. `expected 50, got 49.8`).
  - If you wanted to fix it, you’d fix it
  - If for _reasons_ that’s good enough, you could “lock in” the value (e.g. round it down to `49`) and call it a day

However, even this is proving difficult to easily interpolate between the bronze and silver lookup tables (and I can’t find gold?), as the example tables and calculator only have partial data (and I’m not sure how they’re calculated).

Anyway, this is the only blocker: testing font size + weight. If it’s still a completely undecided area, I’ll punt and we’ll ship the initial version without font testing. But I plan on spending at least a little more time here before giving up.

Reply:
Found a little validation in the [WCAG 3 working draft](https://w3c.github.io/silver/guidelines/#conformance-levels) (emphasis added):

> **Gold** level identifies measures we want to include for those organizations that do achieve Silver so that some can stand out as exemplary, cutting edge, and role models. There are a number of ideas that **will be developed further once more of the conformance structure is solidified.**